Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableAssetHandler.cpp

```diff
@@ -91,4 +94,40 @@ namespace AzFramework
         AZ::Uuid subIdHash = AZ::Uuid::CreateData(id.data(), id.size());
         return azlossy_caster(subIdHash.GetHash());
     }
+
+    void SpawnableAssetHandler::ResolveEntityAliases(
+        Spawnable* spawnable,
+        const AZ::Data::Asset<AZ::Data::AssetData>& asset,
+        AZStd::chrono::milliseconds streamingDeadline,
+        AZ::IO::IStreamerTypes::Priority streamingPriority,
+        const AZ::Data::AssetFilterCB& assetLoadFilterCB)
+    {
+        Spawnable::EntityAliasVisitor aliases = spawnable->TryGetAliases();
+        AZ_Assert(aliases.IsSet(), "Newly created Spawnable '%s' was already locked.", asset.GetHint().c_str());
+        if (aliases.HasAliases())
+        {
+            AZ_Assert(
+                AZStd::is_sorted(
+                    aliases.begin(), aliases.end(),
+                    [](const Spawnable::EntityAlias& lhs, const Spawnable::EntityAlias& rhs)
+                    {
+                        return lhs.HasLowerIndex(rhs);
+                    }),
+                "Spawnable '%s' has an unsorted entity alias list.", asset.GetHint().c_str());
+
+            SpawnableAssetEventsBus::Broadcast(
+                &SpawnableAssetEvents::OnResolveAliases, aliases, spawnable->GetMetaData(), spawnable->GetEntities());
+
+            aliases.Optimize();
```

Comment:
Might be worth a comment just to point out that this is a no-op if the alias list isn't dirty after `OnResolveAliases`